boot/flags.go

```diff
@@ -280,3 +282,60 @@ func setNextBootFlags(dev Device, rootDir string, flags []string) error {
 
 	return m.Write()
 }
+
+// RunModeRootfs returns where the run mode root filesystem is mounted for the
+// given mode.
+// For run mode, it's just "/run/mnt/data".
+// For install mode it's "/run/mnt/ubuntu-data".
+// For recover mode it's either "/host/ubuntu-data" or "" if that is not
+// mounted. Note that, for recover  mode, this function only returns a non-empty
+// return value if the partition is mounted and trusted, there are certain
+// corner-cases where snap-bootstrap in the initramfs may have mounted
+// ubuntu-data in an untrusted manner, but for the purposes of this function
+// that is ignored.
+// This is primarily meant to be consumed by "snap{,ctl} system-mode".
+func RunModeRootfs(mode string) (string, error) {
```

Comment:
Not quite sure about the name, it's not rootfs after all, but more like system's persistent data? How about `PersistentDataWhenInMode(mode string)` ?

Reply:
That's fair, I think that's a good name, but I'd like to hear from @pedronis before I push any renames to save on GCE money :-D

Reply:
we don't really use the term persistent data for this anywhere sofar.  HostWritableWhenInMode or HostUbuntuDataWhenInMode ?

Reply:
how about `HostUbuntuDataForMode` ?

Reply:
I went with my suggestion